### ISSUE TITLE
Problem with PUT IMAGE on CPC with ugBasic 1.13.2

### ISSUE THREAD

**Issue by ldir-hector** [spotlessmind1975/ugbasic#594]
I have a problem with the last version of ugBasic : 

```
BITMAP ENABLE (160,200,16)
CLS
plancheSprite = LOAD IMAGE("explosion_sprites.png")
PUT IMAGE plancheSprite AT 0,0 WITH TRANSPARENCY
```

The image is still supposed to look like this : 
![image](https://user-images.githubusercontent.com/118670680/230664360-8338c411-3097-429b-a2da-f87c7fe86cb9.png)

But now I got this in the emulator : 
![image](https://user-images.githubusercontent.com/118670680/230664429-d5a258e7-6412-459c-8ffd-67aa824412aa.png)

It was working in the previous version.
If I remowe WITH TRANSPARENCY, the result is the same.

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector, and **thank you for your bug report!**

I believe the problem is older than the current version. In fact, if you compile the source with version 1.13.1, you get this result which is wrong:

![image](https://user-images.githubusercontent.com/11718606/230677107-c2fa7d54-89ff-4eff-aade-abbcdabcb28b.png)

I will investigate further.

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector , I share here some reflections and steps I have taken to find the problem, so that they can also be useful for those who will read us. 

![image](https://user-images.githubusercontent.com/11718606/230707946-8c30133c-4d8c-4f50-9ced-41b18d29521c.png)

First of all enable the generation of **additional information** when compiling. This option slows down the compilation but allows you to have very interesting details on the outcome of the automatic algorithms.

![image](https://user-images.githubusercontent.com/11718606/230707967-a607afc4-34a5-4dc1-89d7-c8bd0f779dd0.png)

Now ask to compile (not run!) the program.

![image](https://user-images.githubusercontent.com/11718606/230707984-ba943deb-9751-4679-a22e-489b91812239.png)

After compiling (wait a few moments) a window will open where you can select the **Resources** tab.

![image](https://user-images.githubusercontent.com/11718606/230708000-cda17c23-7233-4374-92c9-f80e8fd19dab.png)

Now click twice on the name of the image, a window will open with the result of the color detection for the construction of the palette.

![image](https://user-images.githubusercontent.com/11718606/230708034-695dcecd-a7cd-4160-b66e-b8a3567d1393.png)

From what can be seen, it would seem that the decoding was done correctly, including the reconstruction of the palette. At this point the problem is not in the common part of the algorithm but in the translation of this result to be used in the cpc environment. 

I did a comparison between version 1.13.1 and version 1.13.2, and there are no major differences -- which means that, if there is a bug, it was already there. Probably, with the introduction of this more "precise" transcoding methodology, we have highlighted a previous problem.

![image](https://user-images.githubusercontent.com/11718606/230708343-fe00a01d-a0ff-45d4-ba34-33572f645bea.png)

The palette seems correct, in terms of hardware indexes. So, the problem is on index transcodification.

I go on with the investigations.

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector , so I was right!
The problem was in the ASM side. 
I fixed it into COLDFIX v1.13.2 [rev. 20230408.1146].

See the actual result:
![image](https://user-images.githubusercontent.com/11718606/230714921-cdc84aee-8c56-487a-a779-47be7676ef98.png)
